lab_5_scrapper/scrapper.py

```diff
@@ -120,45 +254,129 @@ def __init__(self, full_url: str, article_id: int, config: Config) -> None:
         """
         Initializes an instance of the HTMLParser class
         """
-        pass
+        self.full_url = full_url
+        self.article_id = article_id
+        self.config = config
+        self.article = Article(self.full_url, self.article_id)
 
     def _fill_article_with_text(self, article_soup: BeautifulSoup) -> None:
         """
         Finds text of article
         """
-        pass
+        # finds article's text in the unique attribute
+        texts_tag = article_soup.find_all("p")
+        # stores retrieved text in a list
+        final_text = [text.get_text(strip=True) for text in texts_tag]
+        self.article.text = "\n".join(final_text)
 
     def _fill_article_with_meta_information(self, article_soup: BeautifulSoup) -> None:
         """
         Finds meta information of article
         """
-        pass
+        title = article_soup.find("h1")
+
+        # checks that found meta-information exists on the site
+        if title:
+            self.article.title = title.text.strip()
+
+        topic = article_soup.find('a', class_='fn-rubric-a')
+
+        if topic:
+            self.article.topics.append(topic.text.strip())
+
+        date = article_soup.find('div', class_='fn-rubric-link')
+
+        if date:
+            self.article.date = self.unify_date_format(date.text.strip())
+
+        # the name of authors is not defined on the pages
+        self.article.author = ["NOT FOUND"]
 
-    def unify_date_format(self, date_str: str) -> datetime.datetime:
+    @staticmethod
```

Comment:
again staticmethod, why do you use that?

Reply:
done